### PR TITLE
Supermatter Log, Exposed events in inspector and admin nofitication sounds.

### PR DESCRIPTION
**Pull request by MaxIsJoe**
CL: [New] Admins will get a notification when a player attempts to grief the super matter by throwing objects at it. (Parity feature with admin tools on /tg/)
CL: [New] Admins can now investigate who was the last person to touch an item via VV.
CL: [New] Admins will hear a bwoink whenever an alert or prayer notification is unread.
CL: [Improvement] Mappers and designers can now hook events into the pickupable component directly in the inspector without touching code.

### CODE REVIEW

File: UnityProject/Assets/Scripts/Items/LastTouch.cs [hunk 1]
```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using UnityEngine;
+
+namespace Items
+{
+	public class LastTouch : MonoBehaviour
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/LastTouch.cs:6]: uhhhm could this maybe just be part of pickupable?

File: UnityProject/Assets/Scripts/UI/Systems/UIManager.cs [hunk 2]
```diff
@@ -625,4 +626,15 @@ private IEnumerator WaitForStrandedVideoEnd()
 
 		ChatUI.Instance.OpenChatWindow();
 	}
-}
+
+	public static void LogPlayerAction(string incidentTime, PlayerAlertTypes alertType, PlayerInfo perp, string message)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/UI/Systems/UIManager.cs]: feels odd to have this in this component

File: UnityProject/Assets/Scripts/Items/LastTouch.cs (hunk 1 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/Items/LastTouch.cs:6]: The S.O.L.I.D. principal is a principal that says systems should be open to extension and closed to modifications. 

Pickupable should only handle picking up the object. Last Touch should only handle checking who was the last one to touch it. Both components have different responsibilities.

File: UnityProject/Assets/Scripts/UI/Systems/UIManager.cs (hunk 2 above)
Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/UI/Systems/UIManager.cs]: Imo no. And if we were in an ECS environment this would be a common thing to have.

Comment (corp-0) [reply UnityProject/Assets/Scripts/UI/Systems/UIManager.cs]: that can't be more wrong. in ECS you are even more cautelous with where you put stuff. Logging player actions has nothing to do with the UIManager, it isn't a responsibility of the UIManager to pay attention to logs and players actions.

File: UnityProject/Assets/Scripts/Items/LastTouch.cs (hunk 1 above)
Comment (corp-0) [reply UnityProject/Assets/Scripts/Items/LastTouch.cs:6]: recording who last picked up something **IS** part of picking up stuff.

File: UnityProject/Assets/Scripts/UI/Systems/AdminTools/PlayerAlerts.cs [hunk 3]
```diff
@@ -10,6 +10,7 @@ namespace AdminTools
 {
 	public class PlayerAlerts : MonoBehaviour
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/UI/Systems/AdminTools/PlayerAlerts.cs]:
```suggestion
	public class PlayerAlerts : SingletonManager<PlayerAlerts>
```

File: UnityProject/Assets/Scripts/UI/Systems/AdminTools/PlayerAlerts.cs [hunk 4]
```diff
@@ -19,6 +20,16 @@ public class PlayerAlerts : MonoBehaviour
 
 		private readonly List<PlayerAlertData> clientPlayerAlerts = new List<PlayerAlertData>();
 
+		private void Awake()
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/UI/Systems/AdminTools/PlayerAlerts.cs]: I'd say it is better to just use our implementation of singleton monobehaviours

Comment (MaxIsJoe) [reply UnityProject/Assets/Scripts/UI/Systems/AdminTools/PlayerAlerts.cs]: done

**Review by corp-0 — APPROVED** [review]

**Pull request merged**